WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainDetails/ViewModel/RegisterDomainDetailsViewModel.swift

```diff
@@ -243,6 +229,31 @@ class RegisterDomainDetailsViewModel {
         })
     }
 
+    /// Sets the primary domain for  the site.  The result closures are handled internally because we want this method to run
+    /// asynchronously from the rest of the domain purchase flow.   This method failing should not block the user from moving
+    /// forward in the UI.
+    ///
+    /// - Parameters:
+    ///         - domain: the domain to set as primary for the site.
+    ///
+    func setPrimaryDomain(_ domain: String) {
+        let noticeTitle = NSLocalizedString("Domain purchase", comment: "The notice title for the success and failure notices when the App changes the primary site domain to the one the user just purchased.")
+
+        registerDomainDetailsService.changePrimaryDomain(
+            siteID: site.siteID,
+            newDomain: domain,
+            success: {
+                let message = String(format: NSLocalizedString("You site's primary domain was changed to %@", comment: "Notice message shown when the App succeeds in setting the site's primary domain to the domain the user just purchased."), domain)
+                ActionDispatcher.dispatch(NoticeAction.post(Notice(title: noticeTitle, message: message)))
```

Comment:
@Gio2018 - I've added this notice but I'm thinking I should probably remove it on success.  The reason I'm thinking this is because:

1. We previously didn't have it in case of success because the whole process was synchronous.  But this is probably not very interesting for users.
2. For me it comes up when the success screen comes up, and it feels like it's competing for attention.

Thoughts?